integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateLoad.scala

```diff
@@ -415,8 +444,661 @@ test("check load and select for avg double datatype") {
     sql(s"LOAD DATA LOCAL INPATH '$testData' into table maintable")
     sql(s"LOAD DATA LOCAL INPATH '$testData' into table maintable")
     val rows = sql("select age,avg(age) from maintable group by age").collect()
-    sql("create datamap maintbl_douoble on table maintable using 'preaggregate' as select avg(age) from maintable group by age")
+    sql("create datamap maintbl_double on table maintable using 'preaggregate' as select avg(age) from maintable group by age")
     checkAnswer(sql("select age,avg(age) from maintable group by age"), rows)
+    sql("drop table if exists maintable ")
+  }
+
+  def testFunction(): Unit = {
+    // check answer
+    checkAnswer(sql(s"SELECT * FROM main_table_preagg_sum"),
+      Seq(Row(1, 31), Row(2, 27), Row(3, 70), Row(4, 55)))
+    checkAnswer(sql(s"SELECT * FROM main_table_preagg_avg"),
+      Seq(Row(1, 31, 1), Row(2, 27, 1), Row(3, 70, 2), Row(4, 55, 2)))
+    checkAnswer(sql(s"SELECT * FROM main_table_preagg_count"),
+      Seq(Row(1, 1), Row(2, 1), Row(3, 2), Row(4, 2)))
+    checkAnswer(sql(s"SELECT * FROM main_table_preagg_min"),
+      Seq(Row(1, 31), Row(2, 27), Row(3, 35), Row(4, 26)))
+    checkAnswer(sql(s"SELECT * FROM main_table_preagg_max"),
+      Seq(Row(1, 31), Row(2, 27), Row(3, 35), Row(4, 29)))
+
+    testPreAggTable()
+  }
+
+  def testPreAggTable(): Unit = {
+    // check select and match or not match pre-aggregate table
+    checkPreAggTable(sql("SELECT id, SUM(age) FROM main_table GROUP BY id"),
```

Comment:
These scenarios are already covered in TestPreAggregateTableSelection

Reply:
test different data type, also can remove if necessary